src/datasource.js

```diff
@@ -614,24 +623,29 @@ export default class OCIDatasource {
     return vars;
   }
 
-  /** 
+  /**
    * List all variable names optionally filtered by regex or/and type
    * Returns list of names with '$' at the beginning. Example: ['$dimensionKey', '$dimensionValue']
+   *
+   * Updates:
+   * Notes on implementation :
+   * If a custom or constant is in  variables and  includeCustom, default is false.
+   * Hence,the varDescriptors list is filtered for a unique set of var names
   */
   getVariables(regex, includeCustom) {
     const varDescriptors = this.getVariableDescriptors(regex, includeCustom) || [];
-    return varDescriptors.map(item => `$${item.name}`);
+    return [ ...new Set(varDescriptors.map(item => `$${item.name}`))];
```

Comment:
I think the current logic that constant and custom variables should not contain regex.
If custom/constants have to contains regex. We need to move filter logic into getVariableDescriptor.

Reply:
This is done ! Pushing the commit